src/R/Components/Impl/PackageManager/Implementation/RPackageManager.cs

```diff
@@ -173,34 +177,28 @@ internal class RPackageManager : IRPackageManager {
             // Fetching of installed and available packages is done in a
             // separate package query session to avoid freezing the REPL.
             try {
-                using (var sessionToken = await _sessionPool.GetSession()) {
+                var startupInfo = new RHostStartupInfo {
+                    RBasePath = _settings.RBasePath,
+                    CranMirrorName = _settings.CranMirror,
+                    CodePage = _settings.RCodePage
+                };
+
+                using (var eval = await _sessionProvider.BeginEvaluationAsync(startupInfo)) {
                     // Get the repos and libpaths from the REPL session and set them
                     // in the package query session
-                    await EvalRepositoriesAsync(sessionToken.Session, await DeparseRepositoriesAsync());
-                    await EvalLibrariesAsync(sessionToken.Session, await DeparseLibrariesAsync());
+                    var repositories = (await DeparseRepositoriesAsync()).ToRStringLiteral();
+                    await WrapRException(eval.ExecuteAsync($"options(repos=eval(parse(text={repositories})))"));
+                    var libraries = (await DeparseLibrariesAsync()).ToRStringLiteral();
+                    await WrapRException(eval.ExecuteAsync($".libPaths(eval(parse(text={libraries})))"));
 
-                    var result = await WrapRException(queryFunc(sessionToken.Session));
+                    var result = await WrapRException(queryFunc(eval));
                     return result.Select(p => p.ToObject<RPackage>()).ToList().AsReadOnly();
                 }
-            } catch (MessageTransportException ex) {
+            } catch (RHostDisconnectedException ex) {
```

Comment:
I think general guidance on host exception handling in OneNote would be useful. I.e. what is being thrown when and what to handle and what to ignore.